tests/forms/unit.py

```diff
@@ -145,3 +151,285 @@ def test_unit_state():
+        filtered_qs = filtered_qs.filter(
+            store__pootle_path__startswith=cleaned["pootle_path"])
+
+    if cleaned.get("vfolder", None):
+        filtered_qs = filtered_qs.filter(
+            vfolders=cleaned["vfolder"])
+
+    if cleaned.get("filter", None):
+        query_attr = "%s_q" % cleaned["filter"].replace("-", "_")
+        query_method = getattr(unit_filter, query_attr, None)
+        if query_method is not None:
+            filtered_qs = filtered_qs.filter(
+                query_method(cleaned.get('user', request_user)))
+
+    if cleaned.get("checks", None):
+        checks = cleaned.get("checks", None)
+        category = cleaned.get("category", None)
+        if checks is not None:
+            filtered_qs = filtered_qs.filter(
+                qualitycheck__false_positive=False,
+                qualitycheck__name__in=checks)
+        elif category:
+            filtered_qs = filtered_qs.filter(
+                qualitycheck__false_positive=False,
+                qualitycheck__category=category)
+
+    if cleaned.get("modified_since", None):
+        filtered_qs = filtered_qs.filter(
+            submitted_on__gt=cleaned['modified_since'])
+
+    if cleaned.get("month", None):
+        [start, end] = cleaned['month']
+        filtered_qs = filtered_qs.filter(
+            submitted_on__gte=start,
+            submitted_on__lte=end)
+
+    if cleaned.get("search", None) and cleaned.get("sfields", None):
+        filtered_qs = UnitTextSearch(filtered_qs).search(
+            cleaned['search'],
+            cleaned['sfields'],
+            "exact" in cleaned.get("soptions", []))
+
+    assert list(unit_filter.filter_qs(**cleaned)) == list(filtered_qs)
+
+
+@pytest.mark.django_db
+def test_get_units_search(units_search_tests):
+    (search, params, request_user, limit, qs) = units_search_tests
+
+    cleaned = params["cleaned_data"]
+    if not cleaned.get("user", None):
+        cleaned["user"] = request_user
+
+    # the search instance has the qs attr
+    assert list(search.qs) == list(qs)
+
+    # filtered_qs is equiv to calling the the filter_class
+    # with the qs and the kwa
+    filtered_qs = search.filter_class(qs).filter_qs(**cleaned)
+    assert list(search.filtered_qs) == list(filtered_qs)
+    assert search.total == filtered_qs.count()
+
+    # sorted_qs is equiv to calling the the sort_class
+    # with the filtered_qs and the sort kwa
+    sorted_qs = search.sort_class(filtered_qs).sort_qs(
+        cleaned.get("sort_on"), cleaned.get("sort_by"))
+    assert list(search.sorted_qs) == list(sorted_qs)
+
+    # sort field and comparators
+    if sorted_qs.query.order_by:
+        assert search._order_by == sorted_qs.query.order_by[0]
+        assert search.order_by == search._order_by.strip("-")
+    else:
+        assert search._order_by is None
+        assert search.order_by == "pk"
+
+    lte = (
+        cleaned["sort_by"]
+        and cleaned["sort_by"].startswith("-"))
+    if lte:
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Statement seems to have no effect](https://www.codacy.com/app/dwayne/pootle/file/2000838913/issues/source?bid=1990595&fileBranchId=2994326#l395)